src/methods/join.js

```diff
@@ -286,143 +286,107 @@ export class JoinImplementation {
     // effects collected after pnc was constructed
     e: Effects
   ): ForkedAbruptCompletion {
+    let recurse = (x: PossiblyNormalCompletion, xe: Effects): [ForkedAbruptCompletion, Effects] => {
+      let nx = this.replacePossiblyNormalCompletionWithForkedAbruptCompletion(realm, x, ac, e);
+      let nxe = new Effects(nx, xe.generator, xe.modifiedBindings, xe.modifiedProperties, xe.createdObjects);
+      return [nx, nxe];
+    };
+
+    let cloneEffects = () => {
+      // TODO (hermanv) if we use e as is, it ends up being applied twice when the join of the normal
```

Comment:
@hermanventer: is this reuse of your `TODO` logic ok?

Reply:
I'm currently working on this code, so there will be merge conflicts for one of us.